cpp/src/c_api/replicate_edgelist.cpp

```diff
@@ -0,0 +1,157 @@
+  rmm::device_uvector<vertex_t> edgelist_dsts(dst->size_, handle.get_stream());
+
+  raft::copy<vertex_t>(
+    edgelist_srcs.data(), src->as_type<vertex_t>(), src->size_, handle.get_stream());
+  raft::copy<vertex_t>(
+    edgelist_dsts.data(), dst->as_type<vertex_t>(), dst->size_, handle.get_stream());
+
+  std::optional<rmm::device_uvector<weight_t>> edgelist_weights =
+    weights ? std::make_optional(rmm::device_uvector<weight_t>(weights->size_, handle.get_stream()))
+            : std::nullopt;
+
+  if (edgelist_weights) {
+    raft::copy<weight_t>(
+      edgelist_weights->data(), weights->as_type<weight_t>(), weights->size_, handle.get_stream());
+  }
+
+  try {
+    auto gathered_edgelist_srcs = cugraph::c_api::detail::device_allgatherv(
+      handle, raft::device_span<vertex_t const>(edgelist_srcs.data(), edgelist_srcs.size()));
+
+    auto gathered_edgelist_dsts = cugraph::c_api::detail::device_allgatherv(
+      handle, raft::device_span<vertex_t const>(edgelist_dsts.data(), edgelist_dsts.size()));
+
+    rmm::device_uvector<size_t> edge_offsets(2, handle.get_stream());
+    std::vector<size_t> h_edge_offsets{{0, gathered_edgelist_srcs.size()}};
+    raft::update_device(
+      edge_offsets.data(), h_edge_offsets.data(), h_edge_offsets.size(), handle.get_stream());
+
+    // FIXME: Hnadle this case better
+    if (edgelist_weights) {
+      auto gathered_edgelist_weights = cugraph::c_api::detail::device_allgatherv(
+        handle,
+        raft::device_span<weight_t const>(edgelist_weights->data(), edgelist_weights->size()));
+
+      *result = new cugraph::c_api::cugraph_induced_subgraph_result_t{
+        new cugraph::c_api::cugraph_type_erased_device_array_t(gathered_edgelist_srcs, src->type_),
+        new cugraph::c_api::cugraph_type_erased_device_array_t(gathered_edgelist_dsts, dst->type_),
+        new cugraph::c_api::cugraph_type_erased_device_array_t(gathered_edgelist_weights,
+                                                               weights->type_),
+        new cugraph::c_api::cugraph_type_erased_device_array_t(edge_offsets,
+                                                               cugraph_data_type_id_t::SIZE_T)};
+    } else {
+      *result = new cugraph::c_api::cugraph_induced_subgraph_result_t{
+        new cugraph::c_api::cugraph_type_erased_device_array_t(gathered_edgelist_srcs, src->type_),
+        new cugraph::c_api::cugraph_type_erased_device_array_t(gathered_edgelist_dsts, dst->type_),
+        NULL,
+        new cugraph::c_api::cugraph_type_erased_device_array_t(edge_offsets,
+                                                               cugraph_data_type_id_t::SIZE_T)};
+    }
+
+  } catch (std::exception const& ex) {
+    *error = new cugraph::c_api::cugraph_error_t{ex.what()};
+    return CUGRAPH_UNKNOWN_ERROR;
+  }
+
+  return CUGRAPH_SUCCESS;
+}
+
+}  // namespace
+
+// template <typename vertex_t, typename weight_t>
+//  why taking out the 'extern "C"' worked.
```

Comment:
Unsure of the context of this comment.

`extern "C"` is necessary for this to be callable from C code.  Without the `extern "C"` here, the C++ compiler will create the entry point for this function using C++ semantics, so it will mangle the name to include all of the type information.  A regular C compiler doesn't know how to do that.

The `template` construct can't be used on an `extern "C"` call, since templates are not a supported capability in C.

Reply:
> Unsure of the context of this comment.
I was annotating the code with comments (for myself) when I had questions and learning on the process. I figured out "C" doesn't have template and removing the extern "C" will not have C linkage. I forgot to remove this comment  once I figured that out